### PR TITLE
Implement aging report mailing

### PR DESCRIPTION
**Pull request by ehuelsmann**
This PR fixes one of the older regressions we have outstanding. It also builds a foundation for further workflow (state machine) based development of new or re-architected features.

### CODE REVIEW

File: sql/changes/1.9/workflow-email-table.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+   "notify" boolean default false,
+   subject text,
+   body text,
+   sent_date date
+   );
+
+insert into file_class values (8, 'e-mail');
+
+CREATE TABLE file_email (
+       check (file_class=8),
```
Comment (ylavoie) [inline sql/changes/1.9/workflow-email-table.sql:17]: file_class=8? Neumonics would help lisibility & understanding

File: lib/LedgerSMB/Scripts/report_aging.pm [hunk 2]
```diff
@@ -132,15 +132,59 @@ sub generate_statement {
...
+        my $statement = $statements[0];
         $template->render(
             {
                 statements => \@statements,
                 DBNAME     => $request->{company},
             });
 
+        my (@to, @cc, @bcc);
+        # Select billing or regular addresses from the ECA
+        for my $class (15, 12) {
```
Comment (ylavoie) [inline lib/LedgerSMB/Scripts/report_aging.pm:144]: 15, 12?

**Review by ylavoie — COMMENTED** [review]
There is a typo in Implement aging report mailing commit description: "Attachments can be added programmatically or through manually
through uploads."

File: lib/LedgerSMB/Sysconfig.pm [hunk 3]
```diff
@@ -342,6 +343,11 @@ def 'templates_cache',
     default => 'lsmb_templates',
     doc => q{this is a subdir of tempdir, unless it's an absolute path};
 
+def 'workflows',
+    section => 'paths',
+    default => 'workflows',
+    doc => q{};
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Sysconfig.pm:349]: The docstring needs to be completed.

File: sql/changes/1.9/workflow-email-table.sql (hunk 1 above)
Comment (ehuelsmann) [reply sql/changes/1.9/workflow-email-table.sql:17]: True. It is an exact copy of the same code in pg-database. It is the same 8 as the one mentioned above which is being inserted into the file_class table. Values 1 through 7have been taken by other attachment types.
Adding a mnemonic here can't be done: the file isn't loaded through psql

File: lib/LedgerSMB/Scripts/report_aging.pm (hunk 2 above)
Comment (ehuelsmann) [reply lib/LedgerSMB/Scripts/report_aging.pm]: Replaced in 1c87ad1

File: sql/changes/1.9/workflow-email-table.sql (hunk 1 above)
Comment (ehuelsmann) [reply sql/changes/1.9/workflow-email-table.sql:17]: Thinking about it, I don't know how to resolve this comment. Unless you have any ideas, I'm afraid I'll have to leave it unresolved.

File: lib/LedgerSMB/Sysconfig.pm (hunk 3 above)
Comment (ehuelsmann) [reply lib/LedgerSMB/Sysconfig.pm]: Resolved in 8a342cf

**Comment by ehuelsmann** [comment]
> There is a typo in Implement aging report mailing commit description: "Attachments can be added programmatically or through manually
> through uploads."

Resolved with the latest push.

**Pull request merged**